src/main/java/eod/warObject/character/concrete/transparent/IntelligenceVendor.java

```diff
@@ -0,0 +1,163 @@
+package eod.warObject.character.concrete.transparent;
+
+import eod.GameObject;
+import eod.Party;
+import eod.Player;
+import eod.card.abstraction.summon.SummonCard;
+import eod.card.concrete.summon.IntelligenceVendorSummon;
+import eod.effect.Effect;
+import eod.effect.EffectExecutor;
+import eod.event.Event;
+import eod.event.RoundEndEvent;
+import eod.event.relay.EventReceiver;
+import eod.event.relay.StatusHolder;
+import eod.param.PointParam;
+import eod.warObject.Status;
+import eod.warObject.WarObject;
+import eod.warObject.character.abstraction.Character;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static eod.effect.EffectFunctions.GiveStatus;
+
+public class IntelligenceVendor extends Character {
+    public IntelligenceVendor(Player player) {
+        super(player, 3, 0, Party.TRANSPARENT);
+        registerReceiver(new OwnedAbilities());
+    }
+
+    @Override
+    public SummonCard getSummonCard() {
+        SummonCard c = new IntelligenceVendorSummon();
+        c.setPlayer(player);
+        return c;
+    }
+
+    @Override
+    public String getName() {
+        return "情報販";
+    }
+
+    @Override
+    public ArrayList<Point> getAttackRange() {
+        return player.getBoard().allSpaces(new Point(-1, 0), new PointParam());
+    }
+
+    @Override
+    public void attack(EffectExecutor executor) {
+        super.attack(executor);
+        try {
+            Point p = player.selectPosition(getAttackRange());
+            WarObject object = player.getBoard().getObjectOn(p.x, p.y);
+
+            new AttackEffectLock(object);
```

Comment:
Please move register outside of a constructor.

Reply:
solved